File: ui/Components/Admin/AddUser/Dtos/userDto.ts
import { CrudTimeDto, initialTimeDto, TimeDto } from "../../../../Helpers/DataGrid/CrudTimeDto";
import { initialStoreDto, StoreDto, } from "../../../Stores/Dtos/storeDto";
import { initialRoleDto, RoleDto } from "../../AddRole/Dtos/roleDto";

export interface UserDto extends CrudTimeDto {
    id: number;
    userName: string;
    password: string;
    firstName: string;
    lastName: string;
    gender: string;
    email: string
    phoneNumber: string
}

export interface AddUserDto extends UserDto {
    roleDto: RoleDto;
    isActive: boolean;
    storeDto: StoreDto
    shiftTime: TimeDto
}

export interface DeleteUsersDto extends AddUserDto { }

export interface GetUserDto extends AddUserDto { }

export const initialUserDto: UserDto = {
    id: 0,
    userName: "",
    password: "",
    firstName: "",
    lastName: "",
    gender: "",
    createTime: null,
    deleteTime: null,
    updateTime: null,
    email: "",
    phoneNumber: ""
}
export const initialAddUserDto: AddUserDto = {
    roleDto: initialRoleDto,
    isActive: true,
    id: 0,
    userName: "",
    password: "",
    firstName: "",
    lastName: "",
    gender: "",
    email: "",
    storeDto: initialStoreDto,
    shiftTime: initialTimeDto,
    phoneNumber: ""
}
export interface ForgottenPasswordDto {
    email: string;
    emailConfirmNumber: string;
    userName: string;
    password: string;
}
export const initialForgottenPasswordDto: ForgottenPasswordDto = {
    email: "",
    emailConfirmNumber: "",
    userName: "",
    password: ""
}

export interface HelperTextDto {
    userName?: boolean,
    password?: boolean,
    firstName?: boolean,
    lastName?: boolean,
    gender?: boolean,
    email?: boolean,
    phoneNumber?: boolean,
    roleDto?: boolean,
    storeDto?: boolean,
    startDate?: boolean,
    endDate?: boolean,
}
export const initialHelperTextDto: HelperTextDto = {
    userName: false,
    password: false,
    firstName: false,
    lastName: false,
    gender: false,
    email: false,
    phoneNumber: false,
    roleDto: false,
    storeDto: false,
    startDate: false,
    endDate: false
}

File: api/Bussiness/Services/UserService/Dtos/UserDto.cs
using Bussiness.Helper.Dtos;
using Bussiness.Services.RoleService.Dtos;
using Bussiness.Services.Stores.Dtos;

namespace Bussiness.Services.UserService.Dtos
{
    public class UserDto : CrudTimeDto
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Gender { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
    }
  
    public class AddUserDto : UserDto
    {
        public TimeDto ShiftTime { get; set; } // Personel vardiya saatleri 
        public RoleDto RoleDto { get; set; }
        public bool IsActive { get; set; }
        public StoreDto StoreDto { get; set; }
    }
    public class DeleteUsersDto : AddUserDto
    {
    }
    public class GetUserDto : AddUserDto
    {
        public bool IsHaveBarcode { get; set; }
    }
    public class ForgottenPasswordDto 
    {
        public string Email { get; set; }
        //public bool EmailConfirmStatus { get; set; } // Emaile gönderilen butonun tıklanma durumu
        public string EmailConfirmNumber { get; set; }
        public string? UserName { get; set; }
        public string? Password { get; set; }
    }

    public class UserEntryExitLoginDto
    {
        public UserDto UserDto { get; set; }
        public bool? IsApproval { get; set; }  // Admin giriş - çıkış onayı
    }
}
